File: src/main/java/io/github/trainb0y/simplehud/elements/TimeElement.java
package io.github.trainb0y.simplehud.elements;

import net.minecraft.client.MinecraftClient;

import java.util.List;

public class TimeElement extends Element {
    public TimeElement() {
        super();
    }

    public TimeElement(int x, int y, boolean enabled) {
        super(x, y, enabled);
    }

    public Object[] getArgs(MinecraftClient client) {
        int hours = (int) ((client.world != null ? client.world.getTimeOfDay() : 0) / 1000 + 6) % 24;
        int minutes = (int) ((client.world != null ? client.world.getTimeOfDay() : 0) % 1000) * 60 / 1000;
        return List.of(hours, minutes).toArray();
    }

    @Override
    public String getKey() {
        return "element.simplehud.time";
    }
}

File: src/main/kotlin/io/github/trainb0y/fabrihud/elements/TimeElement.kt
package io.github.trainb0y.fabrihud.elements

import net.minecraft.client.MinecraftClient

// HUD element displaying 24-hour ingame time
class TimeElement : TextElement() {
	override fun getArgs(client: MinecraftClient): List<Any?> {
		val hours = ((client.world?.timeOfDay ?: 0) / 1000 + 6).toInt() % 24
		val minutes = ((client.world?.timeOfDay ?: 0) % 1000).toInt() * 60 / 1000
		return listOf(hours, minutes)
	}

	override val key = "element.fabrihud.time"
}